eng/pipelines/templates/BuildAndTest.yml

```diff
@@ -38,8 +38,31 @@ steps:
               /bl:${{ parameters.repoLogPath }}/build.binlog
               $(_OfficialBuildIdArgs)
               $(_InternalBuildArgs)
+              /p:SkipTestProjects=true
       displayName: Build
 
+    - script: ${{ parameters.dotnetScript }}
+              build
+              tests/workloads.proj
+              -p:SkipPackageCheckForTemplatesTesting=true
+      displayName: Prepare sdks for templates testing
+
+    - script: ${{ parameters.buildScript }}
+              -build
+              -restore
+              -test
+              -configuration ${{ parameters.buildConfig }}
+              /bl:${{ parameters.repoLogPath }}/BuildTemplatesTests.binlog
+              $(_OfficialBuildIdArgs)
+              $(_InternalBuildArgs)
+              /p:SkipTests=false
```

Comment:
Why is this needed?

Reply:
This avoids running Restore for the test projects.

Reply:
We will be restoring it later, why do double work?

Reply:
It will restore all the test projects earlier, and only one project earlier. But we could switch it up to that too.